initfiles/bash/etc/systemd/system/CMakeLists.txt

```diff
@@ -0,0 +1,97 @@
+################################################################################
+#    HPCC SYSTEMS software Copyright (C) 2016 HPCC Systems®.
+#
+#    All rights reserved. This program is free software: you can redistribute it
+#    and/or modify
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and #    limitations under the License.
+################################################################################
+
+## Generate and setup instance service files
+set(componentList
+    hpcc-conf
+    thor-conf
+    dafilesrv
+    eclagent
+    dali
+    dfuserver
+    eclccserver
+    eclscheduler
+    esp
+    roxie
+    sasha
+    thor
+    thorslave
+    backupnode)
+
+set(dafilesrv "dafilesrv" "-L ${LOG_PATH} -I %i -D" "")
+set(dali "daserver" "--daemon %i" "dafilesrv.service")
+set(dfuserver "dfuserver" "--daemon %i" "dafilesrv.service")
+set(eclagent "agentexec" "--daemon %i" "dafilesrv.service")
+set(eclccserver "eclccserver" "--daemon %i" "dafilesrv.service")
+set(eclscheduler "eclscheduler" "--daemon %i" "dafilesrv.service")
+set(esp "esp" "--daemon %i" "dafilesrv.service")
+set(roxie "roxie" "--topology=RoxieTopology.xml --logfile --restarts=2 --stdlog=0 --daemon %i" "dafilesrv.service")
+set(sasha "saserver" "--daemon %i" "dafilesrv.service" "Metadata in-memory store.")
+set(thor "thormaster_lcr" "--daemon %i MASTER=\${THORMASTER}:\${THORMASTERPORT}" "dafilesrv.service")
+set(thorslave "thorslave_lcr" "master=\${THORMASTER}:\${THORMASTERPORT} slave=.:\${SLAVEPORT} slavenum=\${SLAVENUM} logDir=${LOG_PATH}/\${THORNAME}" "dafilesrv.service")
+set(backupnode "backup.sh" "" "")
+
+foreach(component ${componentList})
+    list(GET "${component}" 0 processName)
+    list(GET "${component}" 1 processArgs)
+    list(GET "${component}" 2 dependencies)
+    if("${component}" STREQUAL "thor" OR "${component}" STREQUAL "thorslave" OR "${component}" STREQUAL "hpcc-conf" OR "${component}" STREQUAL "thor-conf" OR "${component}" STREQUAL "backupnode")
+        configure_file(${component}@instance.service.in ${component}@.service @ONLY)
+    else()
+        configure_file(component@instance.service.in ${component}@.service @ONLY)
```

Comment:
Could you comment on this if / else condition and the effect the 2 different configure_file lines have ?

Reply:
component@instance.service.in is generic systemd service file. ${component}@instance.service.in is particular service file listed in if condition.

Reply:
Ming is correct.  There is a default service file template that I've created, and then some special service files for things like thor/thorslaves that have extra steps.